### ISSUE TITLE
Stats verb should return an exception on invalid regex

### ISSUE THREAD

**Issue by sitaram-kalluri** [atsign-foundation/at_server#1296]
### Describe the bug

When the stats verb is supplied with an invalid regex, the Format Exception is suppressed and nothing is returned to the client.

In the context of sync, when fetching the latest server commit id with regex, if apps supply an invalid regex, the server does not respond with an exception; simply writes to the server logs. Because of the timeout logic, the client will receive an AtTimeOutException, but the underlying reason is unknown.

### Steps to reproduce

1. Run a stats verb with to fetch the latest server commit id with an invalid regex: stats:3:buzz)
2. No response is sent back to the client.

### Expected behaviour

When an invalid regex is supplied, the stats verb should throw an exception back to the client.

### Additional context

Secondary Server logs:

```
SEVERE|2023-04-03 15:20:25.144040|StatsResponseHandler|FormatException: Unmatched ')'>buzz)
```
No response to the OpenSSL terminal
```
@cram:f36f0....
data:success
@sitaram@stats:3:>buzz)

```

**Comment by sitaram-kalluri** [comment]
The issue occurs when an invalid regex is supplied to fetch the localCommitId with regex in the listen method of statsNotificationListener, in the sync service. The call to fetch the localCommitId is triggered upon receiving the stats notification from the server. Since the listen method is an asynchronous function, the exception thrown is not propagated to the app.

Below is the approach to fix the issue:
  * via SyncProgressListener
     * Introduce AtClientException in SyncProgress class.
     * Populate the AtClientException in SyncProgressListener when the exception occurs.
     * Stop and set sync service as the failure and inform the app.

**Comment by sitaram-kalluri** [comment]
Work done in PR: 60, The changes are completed and pushed to [invalid_sync_regex](https://github.com/atsign-foundation/at_client_sdk/tree/invalid_sync_regex). Yet to raise PR

**Comment by sitaram-kalluri** [comment]
The PR is merged to trunk in at_client SDK : https://github.com/atsign-foundation/at_client_sdk/pull/1052

**Comment by sitaram-kalluri** [comment]
The changes are published in at_client v3.0.61
